snap/validate.go

```diff
@@ -115,8 +116,17 @@ func ValidateApp(app *AppInfo) error {
 		return fmt.Errorf(`"daemon" field contains invalid value %q`, app.Daemon)
 	}
 
+	// Validate app name
+	if app.Name == "" {
+		return fmt.Errorf("snap app name cannot be empty")
+	}
+
+	if !validAppName.MatchString(app.Name) {
+		return fmt.Errorf("invalid snap app name: %q", app.Name)
```

Comment:
Good idea, @zyga, thank you. As long as users don't have to parse regex to figure out what they need to do, I'm happy with it! I used @niemeyer's recommendation in the latest commit.